numpydoc/numpydoc.py

```diff
@@ -278,6 +298,12 @@ def update_config(app, config=None):
             numpydoc_xref_aliases_complete[key] = value
     config.numpydoc_xref_aliases_complete = numpydoc_xref_aliases_complete
 
+    # Processing to determine whether numpydoc_validation_checks is treated
+    # as a blocklist or allowlist
+    if "all" in config.numpydoc_validation_checks:
+        block = deepcopy(config.numpydoc_validation_checks)
+        config.numpydoc_validation_checks = set(ERROR_MSGS.keys()) - block
```

Comment:
I made a silly mistake:
```
set('all') | set(error_ignores)
```
This error of `set('all') == {'a', 'l'}` did not get caught. Could you add some validation after this conditional that every entry in `numpydoc_validation_checks` is in `ERROR_MSGS`? It might save some other people some pain